central/sensor/service/pipeline/nodeindex/pipeline.go

```diff
@@ -47,24 +50,32 @@ type pipelineImpl struct {
 	riskManager   riskManager.Manager
 }
 
-func (p pipelineImpl) OnFinish(_ string) {
+func (p *pipelineImpl) OnFinish(_ string) {
 }
 
-func (p pipelineImpl) Capabilities() []centralsensor.CentralCapability {
+func (p *pipelineImpl) Capabilities() []centralsensor.CentralCapability {
 	return nil
 }
 
-func (p pipelineImpl) Match(msg *central.MsgFromSensor) bool {
+func (p *pipelineImpl) Reconcile(_ context.Context, _ string, _ *reconciliation.StoreMap) error {
+	return nil
+}
+
+func (p *pipelineImpl) Match(msg *central.MsgFromSensor) bool {
 	return msg.GetEvent().GetIndexReport() != nil
 }
 
-func (p pipelineImpl) Run(ctx context.Context, _ string, msg *central.MsgFromSensor, _ common.MessageInjector) error {
+func (p *pipelineImpl) Run(ctx context.Context, _ string, msg *central.MsgFromSensor, injector common.MessageInjector) error {
 	if !env.NodeIndexEnabled.BooleanSetting() || !features.ScannerV4.Enabled() {
 		// Node Indexing only works correctly when both, itself and Scanner v4 are enabled
 		log.Debugf("Skipping node index message (Node Indexing Enabled: %t, Scanner V4 Enabled: %t",
 			env.NodeIndexEnabled.BooleanSetting(), features.ScannerV4.Enabled())
+		// ACK the message to prevent frequent retries
```

Comment:
Just confirming: this protects against a case where Node Indexing is not enabled, but Sensor keeps trying to use it?

Reply:
We send ACK to prevent the Compliance container generating one Node Index per minute (configurable). If node indexing is enabled on secured cluster but disabled in Central (or scanner V4 is missing), then compliance would retry forever for no good reason.

I will improve the comment to make it clearer.